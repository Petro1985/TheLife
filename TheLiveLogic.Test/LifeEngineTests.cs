using System;
using System.Collections.Generic;
using Snapshooter.Xunit;
using TheLiveLogic.Fields;
using Xunit;

namespace TheLiveLogic.Test;

public class LifeEngineTests
{
    [Fact]
    public void LifeLogicTenTurnCheck()
    {
        var map = new EndlessField();
        map.SetCell(new Coord(3, 3), true);
        map.SetCell(new Coord(4, 4), true);
        map.SetCell(new Coord(4, 5), true);
        map.SetCell(new Coord(3, 5), true);
        map.SetCell(new Coord(2, 5), true);

        var theEngine = new LifeEngine();

        for (var i = 0; i < 9; i++)
        {
            theEngine.MakeTurn(map);
        }

        theEngine.MakeTurn(map);
        var newTurnState = map.GetState();
        Snapshot.Match(newTurnState);
    }
}