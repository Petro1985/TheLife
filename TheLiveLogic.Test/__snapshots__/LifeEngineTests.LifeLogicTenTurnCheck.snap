{
  "Survivors": [
    {
      "X": 5,
      "Y": 8
    },
    {
      "X": 6,
      "Y": 6
    },
    {
      "X": 6,
      "Y": 8
    },
    {
      "X": 6,
      "Y": 7
    },
    {
      "X": 4,
      "Y": 7
    }
  ]
}
